sbt_common/storage/src/main/scala/uk/ac/wellcome/storage/s3/S3TypeStore.scala

```diff
@@ -1,26 +1,37 @@
 package uk.ac.wellcome.storage.s3
 
+import java.io.ByteArrayInputStream
+
+import com.amazonaws.services.s3.AmazonS3
 import com.google.inject.Inject
 import com.twitter.inject.Logging
 import io.circe.{Decoder, Encoder}
 import uk.ac.wellcome.utils.GlobalExecutionContext.context
 import uk.ac.wellcome.utils.JsonUtil._
 
 import scala.concurrent.Future
+import scala.io.Source
+
 
 class S3TypeStore[T] @Inject()(
-  stringStore: S3StringStore
-) extends Logging {
+    s3Client: AmazonS3,
+    s3Config: S3Config
+  )(implicit encoder: Encoder[T], decoder: Decoder[T])
```

Comment:
I feel a bit iffy about this – if I’m creating a read-only instance of the TypeStore, why do I also need to supply an encoder?